docs/www/data-transform.md

```diff
@@ -0,0 +1,387 @@
+What happens under the hood, is that we're reading from an immutable Kafka topic, applying a data transformation process and then we producing to another immutable Kafka topic.
+Your data is never actually transformed, but rather read and written again. 
+
+## Enable data transforms
+
+To enable the coprocessor that runs the data transformation you need to enable the engine in the Redpanda configuration file. 
+
+In your `redpanda.yaml` file (which the default path is `/etc/redpanda/redpanda.yaml`) add these **required** flags under `redpanda`.
+
+```yaml
+enable_coproc: true
+developer_mode: true
+enable_idempotence : false 
+```
+
+You can change the port in which the coproc reads from, this is entirely optional. To achieve this add this section:
+
+```yaml
+coproc_supervisor_server:
+    address: 0.0.0.0
+    port: <new port>
+```
+
+The default location of the data transforms log file is: `/var/lib/redpanda/coprocessor/wasm_engine.log`. You can configure the location of this file in `redpanda.yaml` like this: 
+
+```yaml
+coproc_engine:
+  logFilePath: <file_path>
+```
+
+If you opt-in for all the additional configs your `redpanda.yaml` should look similar to this: 
+
+```yaml
+config_file: /etc/redpanda/redpanda.yaml
+node_uuid: mecCncdqAUKFoM2ojvR2Ai86kGWuhiJ95xpNCsKwe6VTgvHoJ
+pandaproxy: {}
+redpanda:
+  admin:
+  - address: 0.0.0.0
+    port: 9644
+  data_directory: /var/lib/redpanda/data
+  developer_mode: true        ## since this is a tech preview you must set developer mode as true
+  enable_coproc: true         ## data transform flag here. make sure it's under redpanda
+  enable_idempotence : false  ## this flag is also mandatory because we're running a tech preview
+  coproc_engine:
+  logFilePath: <file_path>    ## remember to change the file_path
+  coproc_supervisor_server:
+    address: 0.0.0.0          ## if you want, you can change the ip address here
+    port: <new port>          ## remember to change the port here
+  kafka_api:
+  - address: 0.0.0.0
+    port: 9092
+  node_id: 0
+  rpc_server:
+    address: 0.0.0.0
+    port: 33145
+  seed_servers: []
+rpk:
+  coredump_dir: /var/lib/redpanda/coredump
+  enable_memory_locking: false
+  enable_usage_stats: false
+  overprovisioned: false
+  tune_aio_events: false
+  tune_ballast_file: false
+  tune_clocksource: false
+  tune_coredump: false
+  tune_cpu: false
+  tune_disk_irq: false
+  tune_disk_nomerges: false
+  tune_disk_scheduler: false
+  tune_disk_write_cache: false
+  tune_fstrim: false
+  tune_network: false
+  tune_swappiness: false
+  tune_transparent_hugepages: false
+schema_registry: {}
+```
+
+Save your changes and then restart Redpanda.
+
+If you're running it as a service, you can restart it by running this command: 
+
+```bash
+systemctl restart redpanda
+```
+
+## Generating the data transform package
+
+The data transform is packaged in a Node.js project and uses the WASM (WebAssembly) instruction format.
```

Comment:
Wasm is not all caps - it should be Wasm
I think this occurs a few times in this doc.

Reply:
Other players have been using the term interchangeably (WASM and Wasm), but since [the original creators](https://webassembly.org/) refer to it was Wasm it's probably a good idea to change that as well. 

Addressing with the next force push.